### PR TITLE
feat(governance): allow multisig transfer via governance

### PR DESCRIPTION
**Pull request by re1ro**
[AXE-4065]
* [x] `AxelarServiceGovernance`: allowing governance to transfer multisig by a proposal

[AXE-4065]: https://axelarnetwork.atlassian.net/browse/AXE-4065?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: contracts/governance/AxelarServiceGovernance.sol [hunk 1]
```diff
@@ -83,7 +88,12 @@ contract AxelarServiceGovernance is InterchainGovernance, IAxelarServiceGovernan
         _call(target, callData, nativeValue);
     }
 
-    function transferMultisig(address newMultisig) external onlyMultisig {
+    /**
+     * @notice Transfers the multisig address to a new address
+     * @dev Only the current multisig or the governance can call this function
+     * @param newMultisig The new multisig address
+     */
+    function transferMultisig(address newMultisig) external onlyMultisigOrSelf {
```
Comment (milapsheth) [inline contracts/governance/AxelarServiceGovernance.sol:96]: missing coverage

File: test/governance/AxelarServiceGovernance.js [hunk 2]
```diff
@@ -366,12 +366,53 @@ describe('AxelarServiceGovernance', () => {
         );
     });
 
+
+    it('should trasfer multisig by a governance proposal', async () => {
```
Comment (milapsheth) [inline test/governance/AxelarServiceGovernance.js:370]:
```suggestion
    it('should transfer multisig by a governance proposal', async () => {
```

File: test/governance/AxelarServiceGovernance.js [hunk 3]
```diff
@@ -366,12 +366,53 @@ describe('AxelarServiceGovernance', () => {
...
+
+        const tx = await serviceGovernance.executeProposal(serviceGovernance.address, transferCalldata, 0);
+
+        const block = await ethers.provider.getBlock(tx.blockNumber);
+        const executionTimestamp = block.timestamp;
+
+        await expect(tx)
+            .to.emit(serviceGovernance, 'ProposalExecuted')
+            .withArgs(proposalHash, serviceGovernance.address, transferCalldata, 0, executionTimestamp)
+            // .and.to.emit(serviceGovernance, 'MultisigTransferred')
```
Comment (milapsheth) [inline test/governance/AxelarServiceGovernance.js:397]: check the event?

**Review by milapsheth — APPROVED** [review]

File: test/governance/AxelarServiceGovernance.js (hunk 3 above)
Comment (re1ro) [reply test/governance/AxelarServiceGovernance.js]: Yup just fixed that.

File: contracts/governance/AxelarServiceGovernance.sol (hunk 1 above)
Comment (re1ro) [reply contracts/governance/AxelarServiceGovernance.sol:96]: Added

**Pull request merged**